src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import java.io.IOException;
+
+/**
+ * Contains the main logic for responses by Duke.
+ */
+public class Parser {
+    /**
+     * Executes operations on the task list based on given input and
+     * returns the response by the program.
+     *
+     * @param input Input from user.
+     * @param tasks List of tasks.
+     * @return Program response as a String.
+     * @throws DukeException If input is not recognised.
+     */
+    public static String process(String input, TaskList tasks) throws DukeException {
+        if (input.equals("bye")) {
+            return null;
+        }
+
+        StringBuilder output = new StringBuilder();
+        if (input.equals("list")) {
+            output.append("  Tasks in list:\n");
+            for (int i = 0; i < tasks.size(); i++) {
+                output.append("  ").append(i + 1).append(".").append(tasks.get(i)).append("\n");
```

Comment:
Perhaps the use of `String.format()` could have been better in terms of understandability?

Reply:
I agree with the above. While it is possible to understand currently, the use of `String.format()` may improve understandability further.

Reply:
Understood, thanks for the suggestion. Will do this as part of code quality improvements.